docs/source/algo_walkthrough.rst

```diff
@@ -0,0 +1,360 @@
+- Global constraints (also referred to as loop closure constraints or intra submaps contraints) are regularly searched between a new submap and previous nodes that are considered "*close enough*" in space (part of a certain **search window**) and a strong fit (a good match when running scan matching).
+  Intuitively, those "*global ropes*" introduce knots in the structure and firmly bring two strands closer.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.max_constraint_distance
+    POSE_GRAPH.fast_correlative_scan_matcher.linear_search_window
+    POSE_GRAPH.fast_correlative_scan_matcher_3d.linear_xy_search_window
+    POSE_GRAPH.fast_correlative_scan_matcher_3d.linear_z_search_window
+    POSE_GRAPH.fast_correlative_scan_matcher*.angular_search_window
+
+.. note::
+
+    In practice, global constraints can do more than finding loop closures on a single trajectory. They can also align different trajectories recorded by multiple robots but we will keep this usage and the parameters related to "global localization" out of the scope of this document.
+
+To limit the amount of constraints (and computations), Cartographer only considers a subsampled set of all close nodes for constraints building.
+This is controlled by a sampling ratio constant.
+Sampling too few nodes could result in missed constraints and ineffective loop closures.
+Sampling too many nodes would slow the global SLAM down and prevent real-time loop closures.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.sampling_ratio
+
+When a node and a submap are considered for constraint building, they go through a first scan matcher called the ``FastCorrelativeScanMatcher``.
+This scan matcher has been specifically designed for Cartographer and makes real-time loop closures scan matching possible.
+The ``FastCorrelativeScanMatcher`` relies on a "*Branch and bound*" mechanism to work at different grid resolutions and efficiently eliminate incorrect matchings.
+This mechanism is extensively presented in the Cartographer paper presented earlier in this document.
+It works on an exploration tree whose depth can be controled.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher.branch_and_bound_depth
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher_3d.branch_and_bound_depth
+    POSE_GRAPH.constraint_builder.fast_correlative_scan_matcher_3d.full_resolution_depth
+
+Once the ``FastCorrelativeScanMatcher`` has a good enough proposal (above a minimum score of matching), it is then fed into a Ceres Scan Matcher to refine the pose.
+
+.. code-block:: lua
+
+    POSE_GRAPH.constraint_builder.min_score
+    POSE_GRAPH.constraint_builder.ceres_scan_matcher_3d
+    POSE_GRAPH.constraint_builder.ceres_scan_matcher
+
+When Cartographer runs *the optimization problem*, Ceres is used to rearrange submaps according to multiple "*residuals*".
+A residual is essentially a cost function with an associated weight.
```

Comment:
This is wrong, a residual is the result of a cost function - thus I'd say:
`Residuals are calculated using weighted cost functions.`